modules/SofaSparseSolver/src/SofaSparseSolver/CSR_to_adj.h

```diff
@@ -0,0 +1,16 @@
+#ifndef SOFA_CSR_TO_ADJ_H
+#define SOFA_CSR_TO_ADJ_H
+
+#include <SofaSparseSolver/config.h>
+#include <SofaBaseLinearSolver/MatrixLinearSolver.h>
+
+extern "C" {
+#include <metis.h>
+}
+
+namespace sofa::component::linearsolver
+{
+void CSR_to_adj(int n,int * M_colptr,int * M_rowind,type::vector<int>& adj,type::vector<int>& xadj,type::vector<int>& t_adj, type::vector<int>& t_xadj, type::vector<int>& tran_countvec ); // compute the adjency matrix in CSR format from the matrix given in CSR format
+void CSR_to_adj(int n,int * M_colptr,int * M_rowind,type::vector<int>& adj,type::vector<int>& xadj); 
```

Comment:
Both functions content is very similar. Any way to factorize?

Reply:
By the way the LDL solver is implemented, it seems it needs to save the change on the variables used inside the fonction. Maybe by adding a dummy variable that we free after the execution for LU and Cholesky.